src/realizations/coastal/SchismFormulation.cpp

```diff
@@ -0,0 +1,207 @@
+        // RAINRATE - precipitation
+        {"RAINRATE", { SchismFormulation::METEO, "RAINRATE"}},
+        // SFCPRS - surface atmospheric pressure
+        {"SFCPRS", { SchismFormulation::METEO, "PSFC"}},
+        // SPFH2m - specific humidity at 2m
+        {"SPFH2m", { SchismFormulation::METEO, "Q2D"}},
+        // TMP2m - temperature at 2m
+        {"TMP2m", { SchismFormulation::METEO, "T2D"}},
+        // UU10m, VV10m - wind velocity components at 10m
+        {"U10m", { SchismFormulation::METEO, "U2D"}},
+        {"V10m", { SchismFormulation::METEO, "V2D"}},
+
+        /* Input Boundary Conditions */
+        // ETA2_bnd - water surface elevation at the boundaries
+        {"ETA2_bnd", { SchismFormulation::OFFSHORE, "ETA2_bnd"}},
+        // Q_bnd - flows at boundaries
+        {"Q_bnd_source", { SchismFormulation::CHANNEL_FLOW, "Q_bnd_source"}},
+        // Q_bnd - flows at boundaries
+        {"Q_bnd_sink", { SchismFormulation::CHANNEL_FLOW, "Q_bnd_sink"}},
+    };
+
+std::vector<std::string> SchismFormulation::exported_output_variable_names_ =
+    {
+        "ETA2",
+        "VX",
+        "VY",
+        "BEDLEVEL"
+    };
+
+SchismFormulation::SchismFormulation(
+                                     std::string const& id
+                                     , std::string const& library_path
+                                     , std::string const& init_config_path
+                                     , MPI_Comm mpi_comm
+                                     , std::shared_ptr<ProviderType> met_forcings
+                                     , std::shared_ptr<ProviderType> offshore_boundary
+                                     , std::shared_ptr<ProviderType> channel_flow_boundary
+                                     )
+    : CoastalFormulation(id)
+    , meteorological_forcings_provider_(met_forcings)
+    , offshore_boundary_provider_(offshore_boundary)
+    , channel_flow_boundary_provider_(channel_flow_boundary)
+{
+    bmi_ = std::make_unique<models::bmi::Bmi_Fortran_Adapter>
+        (
+         "schism_coastal_formulation"
+         , library_path
+         , init_config_path
+         , /* model_time_step_fixed = */ true
+         , s_schism_registration_function
+         , mpi_comm
+         );
+}
+
+SchismFormulation::~SchismFormulation() = default;
+
+void SchismFormulation::initialize()
+{
+    auto const& input_vars = bmi_->GetInputVarNames();
+
+    for (auto const& name : input_vars) {
+        if (name == "bmi_mpi_comm_handle") {
+            // This is handled at startup by Bmi_Fortran_Adapter
+            continue;
+        }
+
+        if (name == "ETA2_dt") {
+            double eta2_dt = offshore_boundary_provider_->record_duration();
+            bmi_->SetValue(name, &eta2_dt);
+            continue;
+        }
+        if (name == "Q_dt") {
+            double q_dt = channel_flow_boundary_provider_->record_duration();
+            bmi_->SetValue(name, &q_dt);
+            continue;
+        }
```

Comment:
The `update()` function may need to evolve to respect the disparate `dt` values, and SCHISM's potential need to have each of them.